### PR TITLE
ROX-23510: Adding the date-picker to the compound search filter

### PR DESCRIPTION
**Pull request by sachaudh**
## Description

This PR adds the `date-picker` option 📅 to the compound search filter 🔍 

## Screenshots

Screenshot 2024-05-17 at 3 33 39 PM

### CODE REVIEW

File: ui/apps/platform/src/Components/CompoundSearchFilter/components/CompoundSearchFilter.cy.jsx [hunk 1]
```diff
@@ -186,4 +186,35 @@ describe(Cypress.spec.relative, () => {
...
+        cy.get(selectors.attributeSelectItem('Created Time')).click();
+
+        // The date-picker input should be present
+        cy.get('input[aria-label="Filter by date"]').should('exist');
+
+        // Click on the date-picker toggle
+        cy.get('button[aria-label="Filter by date toggle"]').click();
+
+        // Select a month
+        cy.get('div.pf-v5-c-calendar-month__header-month button').click();
```
Comment (sachaudh) [inline ui/apps/platform/src/Components/CompoundSearchFilter/components/CompoundSearchFilter.cy.jsx:207]: PF v5 Date Picker doesn't have aria labels for the month select dropdown. The year input has one. Not sure why they decided not to add one for this though 🤔 . Maybe something worth asking them in the PF channel. Will do early next week

File: ui/apps/platform/cypress.config.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ module.exports = {
             framework: 'create-react-app',
             bundler: 'webpack',
         },
-        viewportHeight: 400,
```
Comment (sachaudh) [inline ui/apps/platform/cypress.config.js:29]: I received the following error in the tests, so I'm upping the viewport height to accommodate. 

Screenshot 2024-05-17 at 3 09 15 PM

I wanted to see if I could append the menu to `document.body` like we do in other cases, but it didn't seem like there was a straightforward way to do it.

File: ui/apps/platform/src/Components/CompoundSearchFilter/components/CompoundSearchFilter.cy.jsx (hunk 1 above)
Comment (dvail) [reply ui/apps/platform/src/Components/CompoundSearchFilter/components/CompoundSearchFilter.cy.jsx:207]: Weird, could just be an accidental omission. 🤔

**Review by dvail — APPROVED** [review]
:shipit:

**Pull request merged**